osu.Game.Rulesets.Osu/Difficulty/Preprocessing/OsuDifficultyHitObject.cs

```diff
@@ -89,17 +89,77 @@ private void setDistances(double clockRate)
             if (lastObject is Slider lastSlider)
             {
                 computeSliderCursorPosition(lastSlider);
-                TravelDistance = lastSlider.LazyTravelDistance * scalingFactor;
+                TravelDistance = 0;
                 TravelTime = Math.Max(lastSlider.LazyTravelTime / clockRate, min_delta_time);
                 MovementTime = Math.Max(StrainTime - TravelTime, min_delta_time);
+                MovementDistance = Vector2.Subtract(lastSlider.TailCircle.StackedPosition, BaseObject.StackedPosition).Length * scalingFactor;
+
+                int repeatCount = 0;
+
+                Vector2 currSliderPosition = ((OsuHitObject)lastSlider.NestedHitObjects[0]).StackedPosition;
+
+                for (int i = 1; i < lastSlider.NestedHitObjects.Count; i++)
+                {
+                    Vector2 currSlider = Vector2.Subtract(((OsuHitObject)lastSlider.NestedHitObjects[i]).StackedPosition, currSliderPosition);
+                    double currSliderLength = currSlider.Length * scalingFactor;
+
+                    if ((OsuHitObject)lastSlider.NestedHitObjects[i] is SliderEndCircle && !((OsuHitObject)lastSlider.NestedHitObjects[i] is SliderRepeat))
+                    {
+                        Vector2 possSlider = Vector2.Subtract((Vector2)lastSlider.LazyEndPosition, currSliderPosition);
+                        if (possSlider.Length < currSlider.Length)
+                            currSlider = possSlider; // Take the least distance from slider end vs lazy end.
+
+                        currSliderLength = currSlider.Length * scalingFactor;
+                    }
+
+                    if ((OsuHitObject)lastSlider.NestedHitObjects[i] is SliderTick)
+                    {
+                        if (currSliderLength > 120)
+                        {
+                            currSliderPosition = Vector2.Add(currSliderPosition, Vector2.Multiply(currSlider, (float)((currSliderLength - 120) / currSliderLength)));
+                            currSliderLength *= (currSliderLength - 120) / currSliderLength;
+                        }
+                        else
+                            currSliderLength = 0;
+                    }
+                    else if ((OsuHitObject)lastSlider.NestedHitObjects[i] is SliderRepeat)
+                    {
+                        if (currSliderLength > 50)
```

Comment:
Why 50?

Reply:
50 is the radius of a circle and produced reasonable results for movement assessment.

Reply:
This should be `const`'d then